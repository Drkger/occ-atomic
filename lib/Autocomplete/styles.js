import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
