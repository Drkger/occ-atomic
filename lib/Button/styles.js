import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

export default {
    btn: {
        alignItems:'flex-start',
        boxSizing:'border-box',
        display:'inline-block',
        marginBottom:'0',
        textAlign:'center',
        whiteSpace:'nowrap',
        verticalAlign:'middle',
        touchAction:'manipulation',
        cursor:'pointer',
        userSelect:'none',
        padding:'0 30px',
        height:'30px',
        lineHeight:'24px',
        fontFamily:fonts.body,
        fontSize:'14px',
        fontWeight:'600',
        position:'relative',
        transition:'0.3s all',
        maxWidth:'100%',
        borderRadius:'5px',
        textDecoration:'none',
        '&:focus': {
            outline:'none'
        },
        '&:hover': {
            textDecoration:'none'
        }
    },
    cont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        pointerEvents:'none'
    },
    loadCont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        position:'absolute',
        top:'0',
        left:'0',
        pointerEvents:'none'
    },
    // Themes
    main: {
        background:colors.pink,
        color:colors.white,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    secondary: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    white: {
        background:'transparent',
        color:colors.white,
        border:`1px solid ${colors.white}`,
        '&:hover': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:'rgba(255,255,255,0.2)',
            borderColor:colors.white
        },
        '&$disabled': {
            background:`transparent !important`,
            color:`rgba(255,255,255,0.6) !important`,
            border:`1px solid rgba(255,255,255,0.6) !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon(['rgba(255,255,255,0.6)'])) : ''
            }
        }
    },
    flat: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid transparent`,
        '&:hover': {
            background:colors.grey3,
            color:colors.pink,
            borderColor:colors.grey3,
        },
        '&:active': {
            background:colors.grey5,
            color:colors.pink,
            borderColor:colors.grey5,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.pink]))
        },
        '&$loading': {
            background:colors.white,
            borderColor:colors.white
        },
        '&$disabled': {
            background:'transparent !important',
            color:`${colors.grey1} !important`,
            border:`1px solid transparent !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.grey1])) : ''
            },
            '& $loadIcon': {
                background: icons.base(icons.loading.icon([colors.grey1]))
            },
        }
    },
    blue: {
        background:colors.blue,
        color:colors.white,
        border:`1px solid ${colors.blue}`,
        '&:hover': {
            background:colors.blueH,
            color:colors.white,
            borderColor:colors.blueH,
        },
        '&:active': {
            background:colors.blueA,
            color:colors.white,
            borderColor:colors.blueA,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.blueA,
            borderColor:colors.blueA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    loading: {
        pointerEvents:'none',
        '& $cont': {
            visibility:'hidden'
        }
    },
    disabled: {
        background:colors.grey1,
        color:colors.white,
        border:`1px solid ${colors.grey1}`,
        cursor:'default',
        pointerEvents:'none',
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&:hover, &:active, &:focus': {
            background:colors.grey1,
            color:colors.white,
            border:`1px solid ${colors.grey1}`
        }
    },
    // Sizes
    md: {
        fontSize:'16px',
        height:'40px',
        lineHeight:'28px',
        '& i': {
            width:'28px',
            height:'28px'
        }
    },
    lg: {
        fontSize:'18px',
        height:'46px',
        lineHeight:'32px',
        '& i': {
            width:'32px',
            height:'32px'
        }
    },
    // Block
    block: {
        display:'block',
        width:'100%'
    },
    // Icon
    icon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all'
    },
    '@keyframes rotate': {
        from: {transform: 'rotate(0deg)'},
        to: {transform: 'rotate(360deg)'}
    },
    loadIcon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all',
        animation:'rotate 1s infinite linear',
        background: icons.base(icons.loading.icon())
    }
};