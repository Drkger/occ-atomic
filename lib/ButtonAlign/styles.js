export default {
    sm: {
        marginTop:'19px'
    },
    md: {
        marginTop:'9px'
    },
    lg: {
        marginTop:'3px'
    }
};