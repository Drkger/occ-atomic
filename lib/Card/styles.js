import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import shadows from '../subatomic/shadows';

const styles = {
    card: {
        background:colors.white,
        borderRadius:'5px',
        padding:'30px 20px'
    },
    cardNoPadding: {
        extend: 'card',
        padding:'0'
    }
};

const pushShadows = function() {
    for (let i = 0; i<7; i++) {
        styles[`shadow${i}`] = {
            boxShadow: shadows[`lvl${i}`]
        };
    }
};

pushShadows();

export default styles;