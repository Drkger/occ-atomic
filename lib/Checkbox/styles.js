import colors from '../subatomic/colors';
import spacing from '../subatomic/spacing';
import icons from '../subatomic/icons';

export default {
    cont: {
        paddingTop:spacing.tiny,
        paddingBottom:spacing.tiny,
        boxSizing:'border-box',
        display:'flex',
        alignItems:'center',
        cursor:'pointer',
        outline:'0',
        '&:after': {
            content:'""',
            display:'table',
            clear:'both'
        },
        '&:hover $check:after': {
            background:icons.base(icons.check.icon([colors.grey200]))
        }
    },
    check: {
        width:spacing.base,
        height:spacing.base,
        position:'relative',
        '&:before': {
            content:'""',
            width:spacing.small,
            height:spacing.small,
            borderRadius:4,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            border:`1px solid ${colors.grey200}`,
            background:colors.bgWhite
        },
        '&:after': {
            content:'""',
            width:18,
            height:18,
            marginTop:-1,
            position:'absolute',
            top:'50%',
            left:'50%',
            transform:'translate(-50%, -50%)',
            transition:'0.3s all',
            background:'transparent'
        }
    },
    active: {
        '& $check:after': {
            background:[icons.base(icons.check.icon([colors.prim])), '!important']
        }
    },
    disabled: {
        opacity:0.4,
        pointerEvents:'none'
    },
    label: {
        marginLeft:spacing.tiny,
        cursor:'pointer'
    }
};
