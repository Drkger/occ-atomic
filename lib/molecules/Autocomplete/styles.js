export default {
    autoComplete: {
        position:'relative'
    },
    droplist: {
        position:'absolute',
        zIndex:'2',
        top:'49px',
        left:'0',
        width:'100%'
    },
    labels: {
        marginTop:'-18px',
        width:'100%'
    }
};