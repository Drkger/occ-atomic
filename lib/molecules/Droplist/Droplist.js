import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

class Droplist extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            term: props.term ? props.term.trim() : ''
        };
        this.onClick = this.onClick.bind(this);
        this.renderList = this.renderList.bind(this);
    }

    componentWillReceiveProps(props) {
        this.setState({
            term: props.term.trim()
        });
    }

    onClick(item) {
        const { onClick } = this.props;
        if (onClick) onClick(item);
    }

    compareText(text, term) {
        let textLC = text.toLowerCase().normalize('NFD').replace(/[\u0300-\u036f]/g, "");
        let termLC = term.toLowerCase().normalize('NFD').replace(/[\u0300-\u036f]/g, "");
        let index = textLC.indexOf(termLC);
        return index;
    }
    
    separateText(itemText, index, term) {
        let text = [itemText.substr(0, index), itemText.substr(index, term.length), itemText.substr(index+term.length, itemText.length)];
        return text;
    }
    
    renderList(items) {
        const { term } = this.state;
        const { classes, itemTextKey, itemTextRightKey } = this.props;
        const itemsDOM = items.map(item => {
            let index = this.compareText(item[itemTextKey], term);
            if (index >= 0) {
                let text = this.separateText(item[itemTextKey], index, term);
                return (
                    <div
                        key={item[itemTextKey]}
                        onClick={() => this.onClick(item)}
                        className={classes.item}>
                        {text[0].length ? text[0] : ''}
                        <span className={classes.bold}>{text[1].length ? text[1] : ''}</span>
                        {text[2].length ? text[2] : ''}
                        {item[itemTextRightKey] ? <span className={classes.right}>{item[itemTextRightKey]}</span> : ''}
                    </div>
                );
            }
            return false;
        });
        return itemsDOM;
    }
    
    render() {
        const { term } = this.state;
        const { classes, className, style, id, items, groups, groupNameKey, groupIdKey, groupItemsKey, itemTextKey, itemTextRightKey, itemIdKey, onClick } = this.props;
        return (
            <div className={`${classes.block}${className ? ` ${className}` : ''}`} id={id}>
                {
                    groups ? (
                        items.map(group => {
                            return (
                                <div key={group[groupIdKey]} className={classes.groupCon}>
                                    <div className={classes.group}>{group[groupNameKey]}</div>
                                    {
                                        this.renderList(group[groupItemsKey])
                                    }
                                </div>
                            );
                        })
                    ) : (
                        this.renderList(items)
                    )
                }
            </div>
        );
    }
}

Droplist.defaultProps = {
    groups: false
};

Droplist.propTypes = {
    classes: PropTypes.object,
    className: PropTypes.string,
    id: PropTypes.string,
    style: PropTypes.object,
    items: PropTypes.array,
    term: PropTypes.string,
    onClick: PropTypes.func,
    groups: PropTypes.bool,
    groupNameKey: PropTypes.string,
    groupIdKey: PropTypes.string,
    groupItemsKey: PropTypes.string,
    itemTextKey: PropTypes.string,
    itemTextRightKey: PropTypes.string,
    itemIdKey: PropTypes.string
};

export default Droplist;