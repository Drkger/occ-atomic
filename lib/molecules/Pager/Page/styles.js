import colors from '../../../subatomic/colors';

export default {
    li: {
        display:'inline-block',
        cursor:'pointer',
        padding:'2px 11px',
        marginLeft:'4px',
        marginRight:'4px',
        borderRadius:'13px',
        cursor:'pointer',
        transition:'0.3s all',
        outline:'0',
        '&:hover': {
            background:'#ECECEE'
        }
    },
    active: {
        background:`${colors.blue} !important`,
        color:colors.white
    }
};