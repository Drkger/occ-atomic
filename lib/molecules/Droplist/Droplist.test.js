import React from 'react';
import { shallow } from 'enzyme';
import Droplist from './Droplist';
import styles from './styles';
import DroplistJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Droplist", () => {

    it('matches the snapshot', () => {
        const items = [
            {
                text: 'Administrador',
                textRight: '(222)'
            },
            {
                text: 'Administrador Sr',
                textRight: '(134)'
            },
            {
                text: 'Administrador Jr',
                textRight: '(52)'
            },
            {
                text: 'Administrar',
                textRight: '(6)'
            },
        ];
        const wrapper = shallow(
            <Droplist items={items} itemTextKey="text" itemTextRightKey="textRight" itemIdKey="text" classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("DroplistJSS", () => {

    it('matches the snapshot', () => {
        const items = [
            {
                text: 'Administrador',
                textRight: '(222)'
            },
            {
                text: 'Administrador Sr',
                textRight: '(134)'
            },
            {
                text: 'Administrador Jr',
                textRight: '(52)'
            },
            {
                text: 'Administrar',
                textRight: '(6)'
            },
        ];
        const wrapper = shallow(
            <DroplistJSS items={items} itemTextKey="text" itemTextRightKey="textRight" itemIdKey="text" />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Droplist styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});