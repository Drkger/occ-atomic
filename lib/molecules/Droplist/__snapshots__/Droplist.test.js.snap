// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Droplist matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Droplist
    classes={
      Object {
        "block": "block",
        "bold": "bold",
        "group": "group",
        "groupCon": "groupCon",
        "item": "item",
        "right": "right",
      }
    }
    groupIdKey="id"
    groupItemsKey="items"
    groupNameKey="text"
    groups={false}
    itemIdKey="id"
    itemTextKey="text"
    itemTextRightKey="textRight"
    items={
      Array [
        Object {
          "id": 1,
          "text": "Administrador",
          "textRight": "(222)",
        },
        Object {
          "id": 2,
          "text": "Administrador Sr",
          "textRight": "(134)",
        },
        Object {
          "id": 3,
          "text": "Administrador Jr",
          "textRight": "(52)",
        },
        Object {
          "id": 4,
          "text": "Administrar",
          "textRight": "(6)",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="item"
          onClick={[Function]}
        >
          
          <span
            className="bold"
          >
            
          </span>
          Administrador
          <span
            className="right"
          >
            (222)
          </span>
        </div>,
        <div
          className="item"
          onClick={[Function]}
        >
          
          <span
            className="bold"
          >
            
          </span>
          Administrador Sr
          <span
            className="right"
          >
            (134)
          </span>
        </div>,
        <div
          className="item"
          onClick={[Function]}
        >
          
          <span
            className="bold"
          >
            
          </span>
          Administrador Jr
          <span
            className="right"
          >
            (52)
          </span>
        </div>,
        <div
          className="item"
          onClick={[Function]}
        >
          
          <span
            className="bold"
          >
            
          </span>
          Administrar
          <span
            className="right"
          >
            (6)
          </span>
        </div>,
      ],
      "className": "block",
      "id": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <span
              className="bold"
            >
              
            </span>,
            "Administrador",
            <span
              className="right"
            >
              (222)
            </span>,
          ],
          "className": "item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "bold",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
          "Administrador",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "(222)",
              "className": "right",
            },
            "ref": null,
            "rendered": "(222)",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <span
              className="bold"
            >
              
            </span>,
            "Administrador Sr",
            <span
              className="right"
            >
              (134)
            </span>,
          ],
          "className": "item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "bold",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
          "Administrador Sr",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "(134)",
              "className": "right",
            },
            "ref": null,
            "rendered": "(134)",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <span
              className="bold"
            >
              
            </span>,
            "Administrador Jr",
            <span
              className="right"
            >
              (52)
            </span>,
          ],
          "className": "item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "bold",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
          "Administrador Jr",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "(52)",
              "className": "right",
            },
            "ref": null,
            "rendered": "(52)",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <span
              className="bold"
            >
              
            </span>,
            "Administrar",
            <span
              className="right"
            >
              (6)
            </span>,
          ],
          "className": "item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "bold",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
          "Administrar",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "(6)",
              "className": "right",
            },
            "ref": null,
            "rendered": "(6)",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="item"
            onClick={[Function]}
          >
            
            <span
              className="bold"
            >
              
            </span>
            Administrador
            <span
              className="right"
            >
              (222)
            </span>
          </div>,
          <div
            className="item"
            onClick={[Function]}
          >
            
            <span
              className="bold"
            >
              
            </span>
            Administrador Sr
            <span
              className="right"
            >
              (134)
            </span>
          </div>,
          <div
            className="item"
            onClick={[Function]}
          >
            
            <span
              className="bold"
            >
              
            </span>
            Administrador Jr
            <span
              className="right"
            >
              (52)
            </span>
          </div>,
          <div
            className="item"
            onClick={[Function]}
          >
            
            <span
              className="bold"
            >
              
            </span>
            Administrar
            <span
              className="right"
            >
              (6)
            </span>
          </div>,
        ],
        "className": "block",
        "id": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <span
                className="bold"
              >
                
              </span>,
              "Administrador",
              <span
                className="right"
              >
                (222)
              </span>,
            ],
            "className": "item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "bold",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
            "Administrador",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "(222)",
                "className": "right",
              },
              "ref": null,
              "rendered": "(222)",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <span
                className="bold"
              >
                
              </span>,
              "Administrador Sr",
              <span
                className="right"
              >
                (134)
              </span>,
            ],
            "className": "item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "bold",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
            "Administrador Sr",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "(134)",
                "className": "right",
              },
              "ref": null,
              "rendered": "(134)",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <span
                className="bold"
              >
                
              </span>,
              "Administrador Jr",
              <span
                className="right"
              >
                (52)
              </span>,
            ],
            "className": "item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "bold",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
            "Administrador Jr",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "(52)",
                "className": "right",
              },
              "ref": null,
              "rendered": "(52)",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <span
                className="bold"
              >
                
              </span>,
              "Administrar",
              <span
                className="right"
              >
                (6)
              </span>,
            ],
            "className": "item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "bold",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
            "Administrar",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "(6)",
                "className": "right",
              },
              "ref": null,
              "rendered": "(6)",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Droplist styles matches the snapshot 1`] = `
Object {
  "block": Object {
    "background": "#ffffff",
    "borderRadius": "0 0 5px 5px",
    "boxShadow": "0 1px 14px rgba(0, 0, 0, 0.1)",
    "padding": "5px 0",
  },
  "bold": Object {
    "fontWeight": "700",
  },
  "group": Object {
    "color": "#727272",
    "fontFamily": "'Open Sans', sans-serif",
    "fontSize": "11px",
    "lineHeight": "28px",
    "padding": "0 8px",
  },
  "groupCon": Object {
    "borderBottom": "#ebebeb",
  },
  "item": Object {
    "&:hover": Object {
      "& $right": Object {
        "color": "#ffffff",
      },
      "background": "#1476fb",
      "color": "#ffffff",
    },
    "color": "#727272",
    "cursor": "pointer",
    "fontFamily": "'Open Sans', sans-serif",
    "fontSize": "15px",
    "lineHeight": "28px",
    "padding": "0 8px",
    "transition": "0.1s all",
  },
  "right": Object {
    "color": "#bfbfbf",
    "float": "right",
    "transition": "0.1s all",
  },
}
`;

exports[`DroplistJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Droplist)
    groupIdKey="id"
    groupItemsKey="items"
    groupNameKey="text"
    groups={false}
    itemIdKey="id"
    itemTextKey="text"
    itemTextRightKey="textRight"
    items={
      Array [
        Object {
          "id": 1,
          "text": "Administrador",
          "textRight": "(222)",
        },
        Object {
          "id": 2,
          "text": "Administrador Sr",
          "textRight": "(134)",
        },
        Object {
          "id": 3,
          "text": "Administrador Jr",
          "textRight": "(52)",
        },
        Object {
          "id": 4,
          "text": "Administrar",
          "textRight": "(6)",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "block": "Droplist-block-0-1-1",
        "bold": "Droplist-bold-0-1-6",
        "group": "Droplist-group-0-1-3",
        "groupCon": "Droplist-groupCon-0-1-2",
        "item": "Droplist-item-0-1-4",
        "right": "Droplist-right-0-1-5",
      },
      "groupIdKey": "id",
      "groupItemsKey": "items",
      "groupNameKey": "text",
      "groups": false,
      "itemIdKey": "id",
      "itemTextKey": "text",
      "itemTextRightKey": "textRight",
      "items": Array [
        Object {
          "id": 1,
          "text": "Administrador",
          "textRight": "(222)",
        },
        Object {
          "id": 2,
          "text": "Administrador Sr",
          "textRight": "(134)",
        },
        Object {
          "id": 3,
          "text": "Administrador Jr",
          "textRight": "(52)",
        },
        Object {
          "id": 4,
          "text": "Administrar",
          "textRight": "(6)",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "block": "Droplist-block-0-1-1",
          "bold": "Droplist-bold-0-1-6",
          "group": "Droplist-group-0-1-3",
          "groupCon": "Droplist-groupCon-0-1-2",
          "item": "Droplist-item-0-1-4",
          "right": "Droplist-right-0-1-5",
        },
        "groupIdKey": "id",
        "groupItemsKey": "items",
        "groupNameKey": "text",
        "groups": false,
        "itemIdKey": "id",
        "itemTextKey": "text",
        "itemTextRightKey": "textRight",
        "items": Array [
          Object {
            "id": 1,
            "text": "Administrador",
            "textRight": "(222)",
          },
          Object {
            "id": 2,
            "text": "Administrador Sr",
            "textRight": "(134)",
          },
          Object {
            "id": 3,
            "text": "Administrador Jr",
            "textRight": "(52)",
          },
          Object {
            "id": 4,
            "text": "Administrar",
            "textRight": "(6)",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
